src/dashboard/livetvtuner.js

```diff
@@ -2,31 +2,54 @@ define(["globalize", "loading", "libraryMenu", "dom", "emby-input", "emby-button
+        ApiClient.getJSON(ApiClient.getUrl("LiveTv/TunerHosts/Types")).then(function(types) {
             var selectType = view.querySelector(".selectType");
             selectType.innerHTML = types.map(function(t) {
-                return '<option value="' + t.Id + '">' + t.Name + "</option>"
-            }).join("") + '<option value="other">' + globalize.translate("TabOther") + "</option>", selectType.disabled = null != currentId, selectType.value = "", onTypeChange.call(selectType)
-        })
+                return '<option value="' + t.Id + '">' + t.Name + "</option>";
+            }).join("");
+            selectType.innerHTML += '<option value="other">';
+            selectType.innerHTML += globalize.translate("TabOther");
+            selectType.innerHTML += "</option>";
+
+            selectType.disabled = null != currentId;
+            selectType.value = "";
+            onTypeChange.call(selectType);
+        });
     }
 
     function reload(view, providerId) {
-        view.querySelector(".txtDevicePath").value = "", view.querySelector(".chkFavorite").checked = !1, view.querySelector(".txtDevicePath").value = "", providerId && ApiClient.getNamedConfiguration("livetv").then(function(config) {
-            var info = config.TunerHosts.filter(function(i) {
-                return i.Id === providerId
-            })[0];
-            fillTunerHostInfo(view, info)
-        })
+        view.querySelector(".txtDevicePath").value = "";
+        view.querySelector(".chkFavorite").checked = false;
+        view.querySelector(".txtDevicePath").value = "";
+        if (providerId) {
+            ApiClient.getNamedConfiguration("livetv").then(function(config) {
+                var info = config.TunerHosts.filter(function(i) {
+                    return i.Id === providerId;
+                })[0];
+                fillTunerHostInfo(view, info);
+            });
+        }
     }
 
     function fillTunerHostInfo(view, info) {
-        var selectType = view.querySelector(".selectType"),
-            type = info.Type || "";
-        info.Source && isM3uVariant(info.Source) && (type = info.Source), selectType.value = type, onTypeChange.call(selectType), view.querySelector(".txtDevicePath").value = info.Url || "", view.querySelector(".txtFriendlyName").value = info.FriendlyName || "", view.querySelector(".txtUserAgent").value = info.UserAgent || "", view.querySelector(".fldDeviceId").value = info.DeviceId || "", view.querySelector(".chkFavorite").checked = info.ImportFavoritesOnly, view.querySelector(".chkTranscode").checked = info.AllowHWTranscoding, view.querySelector(".chkStreamLoop").checked = info.EnableStreamLooping, view.querySelector(".txtTunerCount").value = info.TunerCount || "0"
+        var selectType = view.querySelector(".selectType");
+        var type = info.Type || "";
+        if (info.Source && isM3uVariant(info.Source)) {
+            selectType.value = info.Source;
+            onTypeChange.call(selectType);
```

Comment:
This change looks wrong, you're missing a case when `info.Source && isM3uVariant(...)` is false but `info.Type` is non-empty. Previously it was assigned after this check, now it's lost.

Correct code looks like this IMHO:
```js
var type = info.Type || "";
if (info.Source && isM3uVariant(info.Source)) {
    type = info.Source;
}
selectType.value = type;
onTypeChange.call(selectType);
```